Plugin crashes if import is attempted without API Key being set

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

Terraform v1.0.0
on darwin_amd64
+ provider registry.terraform.io/ibm-cloud/ibm v1.26.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_resource_group

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please share a link to the ZIP file.
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

terraform import ibm_resource_group.rg d9fffb9433e54d97ab2306a452a04252
ibm_resource_group.rg: Importing from ID "d9fffb9433e54d97ab2306a452a04252"...
ibm_resource_group.rg: Import prepared!
  Prepared ibm_resource_group for import
ibm_resource_group.rg: Refreshing state... [id=d9fffb9433e54d97ab2306a452a04252]
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->


Stack trace from the terraform-provider-ibm_v1.26.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16c98bf]

goroutine 69 [running]:
github.com/IBM/platform-services-go-sdk/resourcemanagerv2.(*ResourceManagerV2).GetEnableGzipCompression(...)
        github.com/IBM/platform-services-go-sdk@v0.18.12/resourcemanagerv2/resource_manager_v2.go:149
github.com/IBM/platform-services-go-sdk/resourcemanagerv2.(*ResourceManagerV2).GetResourceGroupWithContext(0x0, 0x3928c40, 0xc0001aa048, 0xc000e02730, 0xc000dd2000, 0x46d97f0, 0xc000580d80, 0x0)
        github.com/IBM/platform-services-go-sdk@v0.18.12/resourcemanagerv2/resource_manager_v2.go:315 +0x27f
github.com/IBM/platform-services-go-sdk/resourcemanagerv2.(*ResourceManagerV2).GetResourceGroup(...)
        github.com/IBM/platform-services-go-sdk@v0.18.12/resourcemanagerv2/resource_manager_v2.go:295
github.com/IBM-Cloud/terraform-provider-ibm/ibm.resourceIBMResourceGroupExists(0xc00092e500, 0x343ff80, 0xc000dd2000, 0x0, 0x46dbc00, 0x0)
        github.com/IBM-Cloud/terraform-provider-ibm/ibm/resource_ibm_resource_group.go:242 +0x15b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000a4f7a0, 0x3928c08, 0xc0007261c0, 0xc000e2e150, 0x343ff80, 0xc000dd2000, 0xc0000b4208, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:604 +0x3d5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00000dc08, 0x3928c08, 0xc0007261c0, 0xc000726200, 0xc0007261c0, 0x100b705, 0x32c86c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:575 +0x43b
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc0010814e0, 0x3928cb0, 0xc0007261c0, 0xc000e0e0c0, 0xc0010814e0, 0xc0011fe1b0, 0xc000e1cba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x105
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x336f8c0, 0xc0010814e0, 0x3928cb0, 0xc0011fe1b0, 0xc000e0e060, 0x0, 0x3928cb0, 0xc0011fe1b0, 0xc000e1e000, 0xde)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001052000, 0x3942258, 0xc000582600, 0xc000e10000, 0xc000a9f680, 0x467bc90, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc001052000, 0x3942258, 0xc000582600, 0xc000e10000, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000046230, 0xc001052000, 0x3942258, 0xc000582600, 0xc000e10000)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-ibm_v1.26.0 plugin crashed!


### Expected Behavior

Attempted to import a resource group with terraform import ibm_resource_group.rg d9fffb9433e54d97ab2306a452a04252 but no API key was set.

I would have expected an error message.

### Actual Behavior

Error: Plugin did not respond message and stacktrace above

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

terraform import ibm_resource_group.rg d9fffb9433e54d97ab2306a452a04252

or assume any resource without an apikey being set

### Important Factoids

Setting an apikey environment variable make the import work without error.

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/config.go
+++ b/ibm/config.go
@@ -77,16 +77,13 @@ import (
 	"github.com/IBM-Cloud/bluemix-go/api/globaltagging/globaltaggingv3"
 	"github.com/IBM-Cloud/bluemix-go/api/hpcs"
 	"github.com/IBM-Cloud/bluemix-go/api/iam/iamv1"
-	"github.com/IBM-Cloud/bluemix-go/api/iamuum/iamuumv1"
 	"github.com/IBM-Cloud/bluemix-go/api/iamuum/iamuumv2"
 	"github.com/IBM-Cloud/bluemix-go/api/icd/icdv4"
 	"github.com/IBM-Cloud/bluemix-go/api/mccp/mccpv2"
 	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/catalog"
 	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
-	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
 	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev2/controllerv2"
 	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev2/managementv2"
-	"github.com/IBM-Cloud/bluemix-go/api/schematics"
 	"github.com/IBM-Cloud/bluemix-go/api/usermanagement/usermanagementv2"
 	"github.com/IBM-Cloud/bluemix-go/authentication"
 	"github.com/IBM-Cloud/bluemix-go/bmxerror"
@@ -198,11 +195,9 @@ type ClientSession interface {
 	ICDAPI() (icdv4.ICDServiceAPI, error)
 	IAMAPI() (iamv1.IAMServiceAPI, error)
 	IAMPolicyManagementV1API() (*iampolicymanagement.IamPolicyManagementV1, error)
-	IAMUUMAPI() (iamuumv1.IAMUUMServiceAPI, error)
 	IAMUUMAPIV2() (iamuumv2.IAMUUMServiceAPIv2, error)
 	MccpAPI() (mccpv2.MccpServiceAPI, error)
 	ResourceCatalogAPI() (catalog.ResourceCatalogAPI, error)
-	ResourceManagementAPI() (management.ResourceManagementAPI, error)
 	ResourceManagementAPIv2() (managementv2.ResourceManagementAPIv2, error)
 	ResourceControllerAPI() (controller.ResourceControllerAPI, error)
 	ResourceControllerAPIV2() (controllerv2.ResourceControllerAPIV2, error)
@@ -280,9 +275,6 @@ type clientSession struct {
 	containerRegistryClientErr error
 	containerRegistryClient    *containerregistryv1.ContainerRegistryV1
 
-	stxConfigErr  error
-	stxServiceAPI schematics.SchematicsServiceAPI
-
 	certManagementErr error
 	certManagementAPI certificatemanager.CertificateManagerServiceAPI
 
@@ -304,9 +296,6 @@ type clientSession struct {
 	globalTaggingConfigErrV1  error
 	globalTaggingServiceAPIV1 globaltaggingv1.GlobalTaggingV1
 
-	iamUUMConfigErr  error
-	iamUUMServiceAPI iamuumv1.IAMUUMServiceAPI
-
 	iamUUMConfigErrV2  error
 	iamUUMServiceAPIV2 iamuumv2.IAMUUMServiceAPIv2
 
@@ -316,9 +305,6 @@ type clientSession struct {
 	userManagementErr error
 	userManagementAPI usermanagementv2.UserManagementAPI
 
-	enterprise    *enterprisemanagementv1.EnterpriseManagementV1
-	enterpriseErr error
-
 	icdConfigErr  error
 	icdServiceAPI icdv4.ICDServiceAPI
 
@@ -327,9 +313,6 @@ type clientSession struct {
 
 	resourceControllerConfigErrv2  error
 	resourceControllerServiceAPIv2 controllerv2.ResourceControllerAPIV2
-
-	resourceManagementConfigErr  error
-	resourceManagementServiceAPI management.ResourceManagementAPI
 
 	resourceManagementConfigErrv2  error
 	resourceManagementServiceAPIv2 managementv2.ResourceManagementAPIv2
@@ -591,11 +574,6 @@ func (sess clientSession) IAMPolicyManagementV1API() (*iampolicymanagement.IamPo
 	return sess.iamPolicyManagementAPI, sess.iamPolicyManagementErr
 }
 
-// IAMUUMAPI provides IAM UUM APIs ...
-func (sess clientSession) IAMUUMAPI() (iamuumv1.IAMUUMServiceAPI, error) {
-	return sess.iamUUMServiceAPI, sess.iamUUMConfigErr
-}
-
 // IAMUUMAPIV2 provides IAM UUM APIs ...
 func (sess clientSession) IAMUUMAPIV2() (iamuumv2.IAMUUMServiceAPIv2, error) {
 	return sess.iamUUMServiceAPIV2, sess.iamUUMConfigErrV2
@@ -614,11 +592,6 @@ func (sess clientSession) MccpAPI() (mccpv2.MccpServiceAPI, error) {
 // ResourceCatalogAPI ...
 func (sess clientSession) ResourceCatalogAPI() (catalog.ResourceCatalogAPI, error) {
 	return sess.resourceCatalogServiceAPI, sess.resourceCatalogConfigErr
-}
-
-// ResourceManagementAPI ...
-func (sess clientSession) ResourceManagementAPI() (management.ResourceManagementAPI, error) {
-	return sess.resourceManagementServiceAPI, sess.resourceManagementConfigErr
 }
 
 // ResourceManagementAPIv2 ...
@@ -938,22 +911,24 @@ func (c *Config) ClientSession() (interface{}, error) {
 		session.pushServiceClientErr = errEmptyBluemixCredentials
 		session.appConfigurationClientErr = errEmptyBluemixCredentials
 		session.kmsErr = errEmptyBluemixCredentials
-		session.stxConfigErr = errEmptyBluemixCredentials
 		session.cfConfigErr = errEmptyBluemixCredentials
 		session.cisConfigErr = errEmptyBluemixCredentials
 		session.functionConfigErr = errEmptyBluemixCredentials
 		session.globalSearchConfigErr = errEmptyBluemixCredentials
 		session.globalTaggingConfigErr = errEmptyBluemixCredentials
+		session.globalTaggingConfigErrV1 = errEmptyBluemixCredentials
 		session.hpcsEndpointErr = errEmptyBluemixCredentials
 		session.iamConfigErr = errEmptyBluemixCredentials
-		session.iamUUMConfigErr = errEmptyBluemixCredentials
 		session.iamUUMConfigErrV2 = errEmptyBluemixCredentials
 		session.icdConfigErr = errEmptyBluemixCredentials
 		session.resourceCatalogConfigErr = errEmptyBluemixCredentials
-		session.resourceManagementConfigErr = errEmptyBluemixCredentials
+		session.resourceManagerErr = errEmptyBluemixCredentials
 		session.resourceManagementConfigErrv2 = errEmptyBluemixCredentials
 		session.resourceControllerConfigErr = errEmptyBluemixCredentials
 		session.resourceControllerConfigErrv2 = errEmptyBluemixCredentials
+		session.enterpriseManagementClientErr = errEmptyBluemixCredentials
+		session.resourceControllerErr = errEmptyBluemixCredentials
+		session.catalogManagementClientErr = errEmptyBluemixCredentials
 		session.powerConfigErr = errEmptyBluemixCredentials
 		session.ibmpiConfigErr = errEmptyBluemixCredentials
 		session.userManagementErr = errEmptyBluemixCredentials
@@ -992,6 +967,9 @@ func (c *Config) ClientSession() (interface{}, error) {
 		session.cisWAFRuleErr = errEmptyBluemixCredentials
 		session.iamIdentityErr = errEmptyBluemixCredentials
 		session.secretsManagerClientErr = errEmptyBluemixCredentials
+		session.schematicsClientErr = errEmptyBluemixCredentials
+		session.satelliteClientErr = errEmptyBluemixCredentials
+		session.iamPolicyManagementErr = errEmptyBluemixCredentials
 
 		return session, nil
 	}
@@ -1380,7 +1358,7 @@ func (c *Config) ClientSession() (interface{}, error) {
 
 	globalTaggingAPIV1, err := globaltaggingv1.NewGlobalTaggingV1(globalTaggingV1Options)
 	if err != nil {
-		session.globalTaggingConfigErr = fmt.Errorf("Error occured while configuring Global Tagging: %q", err)
+		session.globalTaggingConfigErrV1 = fmt.Errorf("Error occured while configuring Global Tagging: %q", err)
 	}
 	if globalTaggingAPIV1 != nil {
 		session.globalTaggingServiceAPIV1 = *globalTaggingAPIV1
@@ -1392,12 +1370,6 @@ func (c *Config) ClientSession() (interface{}, error) {
 		session.iamConfigErr = fmt.Errorf("Error occured while configuring Bluemix IAM Service: %q", err)
 	}
 	session.iamServiceAPI = iam
-
-	iamuum, err := iamuumv1.New(sess.BluemixSession)
-	if err != nil {
-		session.iamUUMConfigErr = fmt.Errorf("Error occured while configuring Bluemix IAMUUM Service: %q", err)
-	}
-	session.iamUUMServiceAPI = iamuum
 
 	iamuumv2, err := iamuumv2.New(sess.BluemixSession)
 	if err != nil {
@@ -1416,12 +1388,6 @@ func (c *Config) ClientSession() (interface{}, error) {
 		session.resourceCatalogConfigErr = fmt.Errorf("Error occured while configuring Resource Catalog service: %q", err)
 	}
 	session.resourceCatalogServiceAPI = resourceCatalogAPI
-
-	resourceManagementAPI, err := management.New(sess.BluemixSession)
-	if err != nil {
-		session.resourceManagementConfigErr = fmt.Errorf("Error occured while configuring Resource Management service: %q", err)
-	}
-	session.resourceManagementServiceAPI = resourceManagementAPI
 
 	resourceManagementAPIv2, err := managementv2.New(sess.BluemixSession)
 	if err != nil {
@@ -1979,7 +1945,7 @@ func (c *Config) ClientSession() (interface{}, error) {
 	}
 	iamIdentityClient, err := iamidentity.NewIamIdentityV1(iamIdentityOptions)
 	if err != nil {
-		session.vpcErr = fmt.Errorf("Error occured while configuring IAM Identity service: %q", err)
+		session.iamIdentityErr = fmt.Errorf("Error occured while configuring IAM Identity service: %q", err)
 	}
 	if iamIdentityClient != nil && iamIdentityClient.Service != nil {
 		iamIdentityClient.Service.EnableRetries(c.RetryCount, c.RetryDelay)
@@ -2000,7 +1966,7 @@ func (c *Config) ClientSession() (interface{}, error) {
 	}
 	iamPolicyManagementClient, err := iampolicymanagement.NewIamPolicyManagementV1(iamPolicyManagementOptions)
 	if err != nil {
-		session.vpcErr = fmt.Errorf("Error occured while configuring IAM Policy Management service: %q", err)
+		session.iamPolicyManagementErr = fmt.Errorf("Error occured while configuring IAM Policy Management service: %q", err)
 	}
 	if iamPolicyManagementClient != nil && iamPolicyManagementClient.Service != nil {
 		iamPolicyManagementClient.Service.EnableRetries(c.RetryCount, c.RetryDelay)

